src/components/modal/ServiceModal.js

```diff
@@ -0,0 +1,490 @@
+  const [showPassword2, setShowPassword2] = React.useState(false);
+  const [showPassword3, setShowPassword3] = React.useState(false);
+  const [done, setDone] = React.useState(false);
+
+  const [formData, setFormData] = React.useState({
+    profile_name: "",
+    api_key: "",
+    auth_calim_key: "",
+    realm_key: "",
+    number_of_services: "",
+  });
+
+  const {
+    profile_name,
+    api_key,
+    auth_calim_key,
+    realm_key,
+    number_of_services,
+  } = formData;
+
+  const [validateForm, setValidateForm] = React.useState({
+    profile_name_valid: false,
+    api_key_valid: false,
+    auth_calim_valid: false,
+    realm_key_valid: false,
+  });
+
+  const {
+    profile_name_valid,
+    api_key_valid,
+    auth_calim_valid,
+    realm_key_valid,
+  } = validateForm;
+
+  const onChange = (e) => {
+    setError("");
+    setFormData((prevState) => ({
+      ...prevState,
+      [e.target.name]: e.target.value,
+    }));
+  };
+
+  const handleClickShowPasswordAPI = () => setShowPassword((show) => !show);
+
+  const handleMouseDownPasswordAPI = (event) => {
+    event.preventDefault();
+  };
+  const handleClickShowPasswordAuth = () => setShowPassword2((show) => !show);
+
+  const handleMouseDownPasswordAuth = (event) => {
+    event.preventDefault();
+  };
+  const handleClickShowPasswordRealm = () => setShowPassword3((show) => !show);
+
+  const handleMouseDownPasswordRealm = (event) => {
+    event.preventDefault();
+  };
+
+  async function handleAddProfile() {
+    if (
+      profile_name === "" ||
+      api_key === "" ||
+      realm_key === "" ||
+      auth_calim_key === ""
+    ) {
+      setError("Please fill out all fields!");
+      return;
+    }
+    setDone(true);
+
+    const data = {
+      profile_name,
+      api_key,
+      auth_calim_key,
+      realm_key,
+      number_of_services,
+    };
+    await profileService.postProfile(data).then((result) => {
+      console.log(result);
+      setDone(true);
+    });
+  }
+
+  function handleCloseModal() {
+    setServiceModal(false);
+    setShowPassword(false);
+    setShowPassword2(false);
+    setShowPassword3(false);
+  }
+
+  return (
+    <>
+      {!done ? (
+        <div>
+          <Modal
+            open={serviceModal}
+            onClose={() => {
+              handleCloseModal();
+            }}
+            aria-labelledby="modal-modal-title"
+            aria-describedby="modal-modal-description"
+          >
+            <Box sx={style}>
+              <Box sx={cloes}>
+                <Button
+                  onClick={() => handleCloseModal()}
+                  sx={{
+                    display: "flex",
+                    justifyContent: "end",
+                    color: "#A3A4A2",
+                    minWidth: "2rem",
+                    minHeight: "2.5rem",
+                    animationTimeline: 5000,
+                    "&:hover": {
+                      opacity: 1,
+                      boxShadow: 4,
+                    },
```

Comment:
can remove inline styles and replace them with classes ?

Reply:
fixed